cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -1103,7 +1142,42 @@ func newServer(conf rootConfig) (*Server, error) {
 }
 
 func main() {
-	if err := rootCmd().ParseAndRun(context.Background(), os.Args[1:]); err != nil {
+	cmd := rootCmd()
+	if err := cmd.Parse(os.Args[1:]); err != nil {
+		log.Fatal(err)
+	}
+
+	// Visit the flags to determine if debug flag is included and enabled
+	debugEnabled := false
+	cmd.FlagSet.Visit(func(flag *flag.Flag) {
```

Comment:
After our code review pairing session I've made a few changes to make the code more explicit. We still split parse and run and write to the env var, so I will also update the PR description so that I clarify why